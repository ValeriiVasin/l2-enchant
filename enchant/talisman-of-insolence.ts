import { Item } from '../items';
import { EnchantMap } from '../types';

// https://forums.lineage2.com/topic/10049-tower-of-insolence/?do=findComment&comment=76516
export const talismanOfBaiumEnchantMap: EnchantMap = new Map([
  [
    Item.TalismanOfInsolence_1,
    {
      item: Item.TalismanOfInsolence_1,
      cost: new Map([
        [Item.ElementalStone, 40],
        [Item.EnergyOfInsolence, 2],
        [Item.TalismanCrystal, 30],
      ]),
      results: [
        { chance: 20, item: Item.TalismanOfInsolence_1 },
        { chance: 80, item: Item.TalismanOfInsolence_2 },
      ],
    },
  ],
  [
    Item.TalismanOfInsolence_2,
    {
      item: Item.TalismanOfInsolence_2,
      cost: new Map([
        [Item.ElementalStone, 100],
        [Item.EnergyOfInsolence, 3],
        [Item.TalismanCrystal, 45],
      ]),
      results: [
        { chance: 40, item: Item.TalismanOfInsolence_2 },
        { chance: 60, item: Item.TalismanOfInsolence_3 },
      ],
    },
  ],
  [
    Item.TalismanOfInsolence_3,
    {
      item: Item.TalismanOfInsolence_3,
      cost: new Map([
        [Item.ElementalStone, 100],
        [Item.EnergyOfInsolence, 4],
        [Item.TalismanCrystal, 70],
      ]),
      results: [
        { chance: 10, item: Item.TalismanOfInsolence_2 },
        { chance: 40, item: Item.TalismanOfInsolence_4 },
        { chance: 50, item: Item.TalismanOfInsolence_3 },
      ],
    },
  ],
  [
    Item.TalismanOfInsolence_4,
    {
      item: Item.TalismanOfInsolence_4,
      cost: new Map([
        [Item.ElementalStone, 200],
        [Item.EnergyOfInsolence, 5],
        [Item.TalismanCrystal, 100],
      ]),
      results: [
        { chance: 10, item: Item.TalismanOfInsolence_5 },
        { chance: 40, item: Item.TalismanOfInsolence_3 },
        { chance: 50, item: Item.TalismanOfInsolence_4 },
      ],
    },
  ],
  [
    Item.TalismanOfInsolence_5,
    {
      item: Item.TalismanOfInsolence_5,
      cost: new Map([[Item.Bx, 5_000]]),
      results: [{ chance: 100, item: Item.TalismanOfInsolence_6 }],
    },
  ],
]);
