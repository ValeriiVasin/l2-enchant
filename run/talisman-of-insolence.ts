import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.TalismanOfInsolence_1,
  [
    Item.TalismanOfInsolence_2,
    Item.TalismanOfInsolence_3,
    Item.TalismanOfInsolence_4,
    Item.TalismanOfInsolence_5,
    Item.TalismanOfInsolence_6,
  ],
  10_000,
);
