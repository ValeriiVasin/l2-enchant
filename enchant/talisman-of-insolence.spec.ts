import { talismanOfInsolenceEnchantMap } from './talisman-of-insolence';

it('enchant map: talisman of insolence', () => {
  expect(talismanOfInsolenceEnchantMap).toMatchSnapshot();
});
