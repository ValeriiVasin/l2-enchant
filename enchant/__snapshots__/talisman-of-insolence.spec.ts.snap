// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: talisman of insolence 1`] = `
Map {
  "Талисман Дерзости Ур.1" => Object {
    "cost": Map {
      "Камень Стихий" => 40,
      "Энергия Дерзости" => 2,
      "Кристалл для Талисмана" => 30,
    },
    "item": "Талисман Дерзости Ур.1",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Дерзости Ур.1",
      },
      Object {
        "chance": 80,
        "item": "Талисман Дерзости Ур.2",
      },
    ],
  },
  "Талисман Дерзости Ур.2" => Object {
    "cost": Map {
      "Камень Стихий" => 100,
      "Энергия Дерзости" => 3,
      "Кристалл для Талисмана" => 45,
    },
    "item": "Талисман Дерзости Ур.2",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Дерзости Ур.2",
      },
      Object {
        "chance": 60,
        "item": "Талисман Дерзости Ур.3",
      },
    ],
  },
  "Талисман Дерзости Ур.3" => Object {
    "cost": Map {
      "Камень Стихий" => 100,
      "Энергия Дерзости" => 4,
      "Кристалл для Талисмана" => 70,
    },
    "item": "Талисман Дерзости Ур.3",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Дерзости Ур.2",
      },
      Object {
        "chance": 40,
        "item": "Талисман Дерзости Ур.4",
      },
      Object {
        "chance": 50,
        "item": "Талисман Дерзости Ур.3",
      },
    ],
  },
  "Талисман Дерзости Ур.4" => Object {
    "cost": Map {
      "Камень Стихий" => 200,
      "Энергия Дерзости" => 5,
      "Кристалл для Талисмана" => 100,
    },
    "item": "Талисман Дерзости Ур.4",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Талисман Дерзости Ур.5",
      },
      Object {
        "chance": 40,
        "item": "Талисман Дерзости Ур.3",
      },
      Object {
        "chance": 50,
        "item": "Талисман Дерзости Ур.4",
      },
    ],
  },
  "Талисман Дерзости Ур.5" => Object {
    "cost": Map {
      "Кристалл: Ранг B" => 5000,
    },
    "item": "Талисман Дерзости Ур.5",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Талисман Дерзости Ур.6",
      },
    ],
  },
}
`;
